components/load_from_hf_hub/src/main.py

```diff
@@ -13,52 +11,33 @@
 logger = logging.getLogger(__name__)
 
 
-def extract_width(image_bytes):
-    # Decode image bytes to PIL Image object
-    pil_image = Image.open(io.BytesIO(image_bytes))
-    width = pil_image.size[0]
-
-    return np.int16(width)
-
-
-def extract_height(image_bytes):
-    # Decode image bytes to PIL Image object
-    pil_image = Image.open(io.BytesIO(image_bytes))
-    height = pil_image.size[1]
-
-    return np.int16(height)
-
-
 class LoadFromHubComponent(LoadComponent):
-    def load(self, *, dataset_name: str) -> dd.DataFrame:
+    def load(self, *, dataset_name: str, column_name_mapping: dict,
+             image_column_name: t.Optional[str]) -> dd.DataFrame:
         """
         Args:
             dataset_name: name of the dataset to load.
-
+            column_name_mapping: column to map the original column names of the input dataset to
+            image_column_name: the name of a column containing images. Used to convert the columns
+            to a proper format. The column name can be either the original column name or the subset
+            specific name
         Returns:
-            Dataset: HF dataset
+            Dataset: HF dataset.
         """
         # 1) Load data, read as Dask dataframe
         logger.info("Loading dataset from the hub...")
         dask_df = dd.read_parquet(f"hf://datasets/{dataset_name}")
 
         # 2) Rename columns
-        dask_df = dask_df.rename(
-            columns={"image": "images_data", "text": "captions_data"}
-        )
+        dask_df = dask_df.rename(columns=column_name_mapping)
 
         # 3) Make sure images are bytes instead of dicts
-        dask_df["images_data"] = dask_df["images_data"].map(
-            lambda x: x["bytes"], meta=("bytes", bytes)
-        )
-
-        # 4) Add width and height columns
-        dask_df["images_width"] = dask_df["images_data"].map(
-            extract_width, meta=("images_width", int)
-        )
-        dask_df["images_height"] = dask_df["images_data"].map(
-            extract_height, meta=("images_height", int)
-        )
```

Comment:
I'm not completely sure about that one, it might be that the original dataset has this metadata and we're assuming that the user requires it. 

I was thinking about implementing another component that generates image metadata to go around this, it can be based on a conditional arguments (e.g. `estimate_width`: True/False, ..).

Reply:
Yes, downside is that it requires loading the image data into the component. We might be able to do this by only loading the first x bytes, but not sure how this works with different image formats and if this can be done performant with parquet.

Reply:
found this library: https://github.com/shibukawa/imagesize_py
Tested it out and it seems to work pretty well